server.js

```diff
@@ -0,0 +1,291 @@
+          }
+        });
+      } else {
+        resp.writeHead(401, {
+          'Content-Type': 'application/json',
+          'WWW-Authenticate': `Basic realm="Secure Area"`,
+        });
+        resp.end(`<html><body>Not Authorized</body></html>`);
+      }
+    }
+  }
+
+  if (req.method === 'DELETE') {
+    const headerObj = req.headers;
+    const hasAuthHeader = headerObj.hasOwnProperty('authorization');
+    if (!hasAuthHeader) {
+      resp.writeHead(401, {
+        'Content-Type': 'application/json',
+        'WWW-Authenticate': `Basic realm="Secure Area"`,
+      });
+      resp.end(`<html><body>Not Authorized</body></html>`);
+    } else if (hasAuthHeader) {
+      const base64auth = headerObj.authorization.substring(6, headerObj.authorization.length);
+      const decodedAuth = new Buffer.from(base64auth, 'base64').toString();
+      const user = decodedAuth.substring(0, decodedAuth.indexOf(':'));
+      const password = decodedAuth.substring(decodedAuth.indexOf(':') + 1, decodedAuth.length);
+      if (userPasswords.hasOwnProperty(user) && userPasswords[user] === password) {
+        fs.access(`public${req.url}`, function(err) {
+          // file exists
+          if (!err) {
+            // delete file
+            fs.unlink(`public${req.url}`, (err) => {
+              if (err) {
+                throw err;
+              }
+              console.log(`public${req.url} was deleted`);
+            });
+            // remake index.html
+            //req.on('end', function() {
+            fs.readdir('./public', (err, files) => {
+              const elemFiles = files.filter(
+                (file) => file !== '.keep' && file !== '404.html' && file !== 'css' && file !== 'index.html',
+              );
+              // generate li tags for each element file
+              let elemTags = '';
+              let elemLowerCase = '';
+              let elemProperCase = '';
+              let elemSubStr = 0;
+              for (let element = 0; element < elemFiles.length; element++) {
+                elemLowerCase = elemFiles[element].toLowerCase();
+                elemProperCase = elemFiles[element].charAt(0).toUpperCase() + elemFiles[element].slice(1);
+                elemSubStr = elemFiles[element].indexOf('.');
+                elemTags += `      <li>
+        <a href="/${elemLowerCase}">${elemProperCase.substring(0, elemSubStr)}</a>
+      </li>
+`;
+              }
+              const indexTemplate = `<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <title>The Elements</title>
+    <link rel="stylesheet" href="/css/styles.css" />
+  </head>
+  <body>
+    <h1>The Elements</h1>
+    <h2>These are all the known elements.</h2>
+    <h3>These are ${elemFiles.length}</h3>
+    <ol>
+${elemTags}    </ol>
+  </body>
+</html>`;
+              fs.writeFile(`public/index.html`, indexTemplate, 'utf8', function(err) {
+                if (err) {
+                  return console.log(err);
```

Comment:
Handle the err from fs.writeFile and send a proper error back to the user with a status of 500